### PR TITLE
Cleanup all ingress resources

### LINKED ISSUE #361: clean up ci not clean up compute engine resources
cleanup-ci only cleans up health check resources for compute engine.  We have a much larger extent of resource leak.  Please take a look at this image:
graph

Basically we need to clean up orphaned resources plus resources associated with presubmit tests which lasted longer than a given amount of time.

This is my WIP codes for garbage collection, someone could start with it and finish it: [link](https://github.com/gabrielwen/testing/blob/garbage-collection-demo/py/kubeflow/testing/cleanup_ig.py)

Note [here](https://github.com/gabrielwen/testing/blob/fd202a547b62689fed45e4ded2bd51676abc28ab/py/kubeflow/testing/cleanup_ig.py#L31) how I use endpoint name to determine if a resource is bound to presubmit tests.  Also, this script still has a lot to improve, such as finding orphaned resources.

### PR DESCRIPTION
**Pull request by zhenghuiwang**
Fixes #361 

Clean up all ingress resources after certain age:
-  cleanup_forwarding_rules,
-  cleanup_target_http_proxies,
-  cleanup_url_maps,
-  cleanup_instance_groups,

Tested it locally.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/testing/390)

### CODE REVIEW

File: py/kubeflow/testing/cleanup_ci.py [hunk 1]
```diff
@@ -611,7 +787,11 @@ def cleanup_all(args):
          cleanup_service_account_bindings,
          cleanup_workflows,
          cleanup_disks,
+         cleanup_forwarding_rules,
+         cleanup_target_http_proxies,
+         cleanup_url_maps,
          cleanup_backend_services,
+         cleanup_instance_groups,
```
Comment (gabrielwen) [inline py/kubeflow/testing/cleanup_ci.py]: I think this one needs to be after `cleanup_health_checks`.

Comment (zhenghuiwang) [reply py/kubeflow/testing/cleanup_ci.py]: Done.

**Comment by zhenghuiwang** [comment]
> have you tried and see if there is any error because of dependencies?

I don't see much error about. I wouldn't worry about the order of these clean up operations. It is a cron job and the resource will be GCed once the dependencies are GCed in last round.

**Comment by gabrielwen** [comment]
> > have you tried and see if there is any error because of dependencies?
> 
> I don't see much error about. I wouldn't worry about the order of these clean up operations. It is a cron job and the resource will be GCed once the dependencies are GCed in last round.

theoretically yes, but it would take O(n) rounds to GC everything, given n = level of dependencies, right?  we are already witnessing increasing speed of resource consumption, should we make some effort to reduce the time of GC?

**Comment by zhenghuiwang** [comment]
Added a comment on the order of these operations, which now clean all ingress resources in one run. PTAL

**Pull request merged**